localstack-core/localstack/services/stepfunctions/asl/component/state/state_execution/state_task/state_task_lambda.py

```diff
@@ -60,12 +63,20 @@ def _from_error(self, env: Environment, ex: Exception) -> FailureEvent:
             return ex.failure_event
 
         error = "Exception"
-        if isinstance(ex, lambda_eval_utils.LambdaFunctionErrorException):
+        if isinstance(ex, ClientError):
             error_name = CustomErrorName(error)
+            cause = ex.response["Error"]["Message"]
+        elif isinstance(ex, lambda_eval_utils.LambdaFunctionErrorException):
             cause = ex.payload
-        elif isinstance(ex, ClientError):
+            try:
+                cause_object = json.loads(cause)
+                error = cause_object["errorType"]
+            except Exception as ex:
+                LOG.warning(
+                    "Could not retrieve 'errorType' field from LambdaFunctionErrorException object: %s",
+                    ex,
+                )
```

Comment:
nit: Could be worth unifying this function and the above to prevent duplication in both the `StateTaskLambda` and `StateTaskServiceLambda`